test/controllers/ProfileControllerSpec.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import base.SpecBase
+import connectors.TraderProfileConnector
+import models.{NormalMode, TraderProfile}
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.when
+import org.scalatestplus.mockito.MockitoSugar
+import play.api.i18n.Messages
+import play.api.inject.bind
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import uk.gov.hmrc.http.NotFoundException
+import viewmodels.checkAnswers._
+import viewmodels.govuk.summarylist._
+import views.html.ProfileView
+
+import scala.concurrent.Future
+
+class ProfileControllerSpec extends SpecBase with MockitoSugar {
+
+  private lazy val profileRoute          = routes.ProfileController.onPageLoad().url
+  private val mockTraderProfileConnector = mock[TraderProfileConnector]
+
+  private val profileResponse = TraderProfile(
+    "actorId",
+    "UKIMS number",
+    Some("NIRMS number"),
+    Some("NIPHL number")
+  )
+
+  "Profile Controller" - {
```

Comment:
can you do a tests for different types of data (like only mandatory data etc)

Reply:
Unit tests added